clippy_lints/src/matches.rs

```diff
@@ -1033,7 +1033,7 @@ fn check_wild_enum_match(cx: &LateContext<'_>, ex: &Expr<'_>, arms: &[Arm<'_>])
 
     // Accumulate the variants which should be put in place of the wildcard because they're not
     // already covered.
-    let mut missing_variants: Vec<_> = adt_def.variants.iter().collect();
+    let mut missing_variants: Vec<_> = adt_def.variants.iter().filter(|x| !is_hidden(cx, x)).collect();
```

Comment:
I think here it would be good to differentiate between the enums origin. It would make sense to me to also cover hidden/unstable variations if they originate from the same crate. (I couldn't find a check for that in the current implementation)

I'm also debating if that should be configurable, but I believe there is no real reason for that.

Reply:
I don't think this has to be done in this PR. I like this idea though. Can you open a good-first-issue issue for this, please?

Reply:
cc: #7419